### PR TITLE
Fix content disposition

### PR DESCRIPTION
**Pull request by butonic**
This supersedes https://github.com/cs3org/reva/pull/2297 by taking the requested filename from the RequestURI. This better reflects the request intent while keeping paths correct inside reva.

cc @C0rby @ishank011 @labkode

Tested with running the oc-acceptance tests with `DEBUG_ACCEPTANCE_API_CALLS=1` for `BEHAT_FEATURE='tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature:41'`
```
        │ ### AUTH: :
        │ ### REQUEST: GET http://localhost:20080/remote.php/dav/public-files/jZxKoheUyZuPHTx/randomfile.txt
        │ Headers:
        │ Host: localhost:20080
        │ X-Requested-With: XMLHttpRequest
        │ Range: bytes=0-9
        │ Body:
        │ string(0) ""
        │ 
        │ ### END REQUEST
        │ ### RESPONSE
        │ Status: 206
        │ Headers:
        │ Access-Control-Allow-Origin: *
        │ Content-Disposition: attachment; filename*=UTF-8''randomfile.txt; filename="randomfile.txt"
        │ Content-Length: 10
        │ Content-Range: bytes 0-9/11
        │ Content-Security-Policy: default-src 'none';
        │ Content-Type: text/plain
        │ Etag: "ea0f28cd2ec02041e462e7a108450312"
        │ Last-Modified: Mon, 22 Nov 2021 14:14:29 +0000
        │ Oc-Etag: "ea0f28cd2ec02041e462e7a108450312"
        │ Oc-Fileid: ZTFhNzNlZGUtNTQ5Yi00MjI2LWFiZGYtNDBlNjljYTgyMzBkOmpaeEtvaGVVeVp1UEhUeC84NThhNDAxOS1iMDRmLTQ1YmMtODI4MS1iNWVkZjcwYmIxYjI=
        │ Vary: Origin
        │ X-Content-Type-Options: nosniff
        │ X-Download-Options: noopen
        │ X-Frame-Options: SAMEORIGIN
        │ X-Permitted-Cross-Domain-Policies: none
        │ X-Robots-Tag: none
        │ X-Xss-Protection: 1; mode=block
        │ Date: Mon, 22 Nov 2021 14:14:30 GMT
        │ Body:
        │ string(10) "Random dat"
        │ 
        │ ### END RESPONSE
```

### CODE REVIEW

**Review by ishank011 — COMMENTED** [review]
@butonic I'm not sure this would work. Consider the tree `outer-folder>inner-folder>file.txt` For a public link created for `outer-folder`, the public storage provider would return `$token/inner-folder/file.txt` for the file.

A get request would look like `GET /remote.php/dav/public-files/$token/inner-folder/file.txt`, this code will first extract the directory and then the base from the path, so you'll get `inner-folder`, using which you'll look up the storage provider and then fail.

File: internal/http/services/owncloud/ocdav/get.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (s *svc) handleGet(ctx context.Context, w http.ResponseWriter, r *http.Requ
 
 	w.Header().Set(HeaderContentType, info.MimeType)
 	w.Header().Set(HeaderContentDisposistion, "attachment; filename*=UTF-8''"+
-		path.Base(info.Path)+"; filename=\""+path.Base(info.Path)+"\"")
+		path.Base(r.RequestURI)+"; filename=\""+path.Base(r.RequestURI)+"\"")
```
Comment (butonic) [inline internal/http/services/owncloud/ocdav/get.go:127]: @ishank011 the trick is to always take the filename from the request url.

Comment (ishank011) [reply internal/http/services/owncloud/ocdav/get.go:127]: Yes we can keep the change but we shouldn't revert https://github.com/cs3org/reva/pull/2297, that'll break the case I described above

Comment (butonic) [reply internal/http/services/owncloud/ocdav/get.go:127]: we should probably cover this with a test @phil-davis

Comment (butonic) [reply internal/http/services/owncloud/ocdav/get.go:127]: @ishank011 could you check if https://github.com/owncloud/core/blob/762780a23c9eadda4fb5fa8db99eba66a5100b6e/tests/acceptance/features/apiSharePublicLink1/createPublicLinkShare.feature#L184-L221 covers the scenario you described? If not could you rephrase it in gherkin?

@phil-davis could you add a check for the content disposition header to the core testsuite?

Comment (ishank011) [reply internal/http/services/owncloud/ocdav/get.go:127]: @butonic will take a look

**Review by labkode — APPROVED** [review]

**Comment by ishank011** [comment]
@labkode this needs to be reverted. Public link downloads won't work with this, like I described above
